### PR TITLE
Sticky search

### LINKED ISSUE #183: Icon library: search should be sticky
@jeanservaas I believe Chiu and/or Shix had specs or a design. Should sit in a container with 8 or 16px under the search field. Icons scroll underneath. Could have very subtle drop shadow maybe?

Safekeepings can send specs if you cannot find files

### PR DESCRIPTION
**Pull request by vpicone**
Closes #183 

New: filter row is sticky for icons and pictograms, pulled out into it's own component shared between the two pages

Two accommodations for this implementation:
- Wrap category dropdown since no carbon equivalent exists for the icon dropdown
- For medium and large viewports, split the input widths 50/50. This allows for an identical implementation between icons and pictograms, as opposed to bespoke column count.

preview: [pictograms](https://deploy-preview-312--design-language-website.netlify.com/design/language/elements/pictogram-library), [icons](https://deploy-preview-312--design-language-website.netlify.com/design/language/elements/icon-library)

### CODE REVIEW

**Comment by vpicone** [comment]
@jeanservaas the icon dropdown thing used here doesn't exist in carbon for dropdown or multiselect. For multi-select in particular, having the first item in the dropdown be the root 'Categories' with the clear all badge isn't behavior we have in any component.

Considering the number of users for this component on mobile, the amount of effort to create the bespoke component or add this functionality to dropdown upstream would make this project substantially more difficult compared to this implementation.

**Comment by vpicone** [comment]
@jeanservaas do we have a 48px height dropdown? The Search has a 40px variant that matches the dropdown so that's what I grabbed.

**Comment by vpicone** [comment]
@jeanservaas since we don't have an official 48px dropdown, I fudged it here and used the same type tokens as the regular sized search. If this is something we anticipate other folks doing, we should look at implementing other sizes for form inputs.

- Changed sm search to normal search
- Bumped up the size and type token of the dropdown
- Added space above filter row
- Used different columns for icons vs pictograms

**Comment by jeanservaas** [comment]
looking better, a couple things:

- Can we have the drop down field heights also be 48px for the sake of consistency with other component dropdowns? I know we don't technically have the 48px dropdown yet but we need to have these geometries be the same eventually.

image

48px dropdown on type tester has correct geometry:

image

- At the tablet breakpoint, I see that the dropdown width is a quarter of the grid so it aligns with the last card, but it's awkwardly narrow. I'd rather see it go half and half (only at this breakpoint) so it doesn't look so crammed

image


- We talked about this offline... but we need to do something here about the awkward vertical spacing in mobile, the component looks like it's floating. Since there's no content above the search sticky component, let's just butt it up against the header in mobile (but only in mobile). 

image


- @mjabbink says he suggested using the overflow component but changing the icon from a chevron to a filter in order to achieve Pete's spec (below). It would be OK not to have the multi-select... what would be the effort on this? would it still be a rabbit hole?

image

Selected state:

image

**Comment by mjabbink** [comment]
Agree. If we don’t do it now it will never get done. Seems doable but @vpicone should way in on effort.

**Comment by vpicone** [comment]
@jeanservaas @mjabbink the overflow menu is for one time actions/navigation not as a form input, so there's *no sense of a selected item* (as opposed to the dropdown which has a currently selected item). Implementing that behavior (an active, selected item) would be a a pretty significant effort to undertake for just mobile.

**Comment by vpicone** [comment]
bumping @connor-leech for some blurbs above the filter row.

**Comment by vpicone** [comment]
@jeanservaas added dropdown item height override and adjusted the tablet columns for icons.

**Comment by mjabbink** [comment]
I was thinking you could have a select check mark next to the item chosen. In the end it’s what Jeannie/Pete spec’d out anyways.

**Comment by vpicone** [comment]
@mjabbink The original spec isn't possible with Carbon components. We can either make our own component, or heavily modify the dropdown/overflow menu to fit this use case. Either option is going to take a bit, I'm reluctant dive in too deep considering this is for mobile users only.

**Comment by mjabbink** [comment]
I think we just proceed with the stacked search over the drop down in mobile.

File: src/components/SvgLibraries/shared/SvgLibrary.module.scss [hunk 1]
```diff
@@ -10,46 +10,100 @@
...
+}
+
+.filter-row::before {
+  content: '';
+  position: absolute;
+  z-index: z('dropdown');
+  height: 1rem;
+  width: 100%;
+  top: -1rem;
+  background: #f3f3f3;
```
Comment (jnm2377) [inline src/components/SvgLibraries/shared/SvgLibrary.module.scss]: Do we wanna use a token for this?

File: src/components/SvgLibraries/shared/SvgLibrary.module.scss [hunk 2]
```diff
@@ -10,46 +10,100 @@
...
   @include carbon--breakpoint('md') {
-    margin-top: 0;
     width: auto;
   }
 }
 
+.filter-row :global(.bx--dropdown) {
+  // handled by filter-row
+  height: 48px;
+  max-height: 48px;
```
Comment (jnm2377) [inline src/components/SvgLibraries/shared/SvgLibrary.module.scss]: should we use rem or mini-units here?

File: src/components/SvgLibraries/shared/SvgLibrary.module.scss (hunk 1 above)
Comment (vpicone) [reply src/components/SvgLibraries/shared/SvgLibrary.module.scss]: @jnm2377 I don't think there is one :( It'd be like gray-05 or something.

**Comment by vpicone** [comment]
@jeanservaas @mjabbink so are we okay with the current implementation?

https://deploy-preview-312--design-language-website.netlify.com/design/language/elements/icon-library

**Comment by mjabbink** [comment]
Seems good to me. @jeanservaas?

**Pull request merged**